source/usr/lib/mcvirt/auth/auth.py

```diff
@@ -0,0 +1,339 @@
+"""Provide auth class for managing permissions."""
+
+# Copyright (c) 2014 - I.T. Dev Ltd
+#
+# This file is part of MCVirt.
+#
+# MCVirt is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# MCVirt is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with MCVirt.  If not, see <http://www.gnu.org/licenses/>
+
+import os
+import Pyro4
+
+from mcvirt.mcvirt_config import MCVirtConfig
+from mcvirt.exceptions import (UserNotPresentInGroup, InsufficientPermissionsException,
+                               UnprivilegedUserException, InvalidPermissionGroupException,
+                               DuplicatePermissionException)
+from mcvirt.rpc.lock import locking_method
+from mcvirt.rpc.pyro_object import PyroObject
+from mcvirt.auth.permissions import PERMISSIONS, PERMISSION_GROUPS
+
+
+class Auth(PyroObject):
+    """Provides authentication and permissions for performing functions within MCVirt."""
+
+    @staticmethod
+    def check_root_privileges():
```

Comment:
To be honest, I think it should be moved into ``mcvirt.utils`` and used in the NameServer and RpcDaemon classes